### PR TITLE
feat: add api runtime flags

### PR DESCRIPTION
**Pull request by sgollapudi77**

### CODE REVIEW

**Review by sinedied — CHANGES_REQUESTED** [review]
Could you also update the config schema in `schema/`?

**Comment by sgollapudi77** [comment]
> Is this to support the use case where the api runtime is not set in staticwebapp.config.json?

Yes, Yohan. But for this to work the code in the StaticSiteclient needs to be merged.

**Comment by sgollapudi77** [comment]
> Could you also update the config schema in `schema/`?

Done. Please check now.

**Review by sinedied — APPROVED** [review]
Thanks!
Could you rebase your branch with latest changes from `ga` so I can merge it?

**Comment by sgollapudi77** [comment]
> Thanks! Could you rebase your branch with latest changes from `ga` so I can merge it?

Done

File: src/cli/commands/deploy.ts [hunk 1]
```diff
@@ -31,6 +31,8 @@ export default function registerCommand(program: Command) {
     .option("--app-location <path>", "the folder containing the source code of the front-end application", DEFAULT_CONFIG.appLocation)
     .option("--api-location <path>", "the folder containing the source code of the API application", DEFAULT_CONFIG.apiLocation)
     .option("--output-location <path>", "the folder containing the built source of the front-end application", DEFAULT_CONFIG.outputLocation)
+    .option("--api-language <apiLanguage>", "the runtime language of the function")
```
Comment (sinedied) [inline src/cli/commands/deploy.ts:34]: Just caught that it's also missing default values and environment variables, could you add it?

**Review by sinedied — CHANGES_REQUESTED** [review]

**Comment by sgollapudi77** [comment]
Closes #519

**Comment by sulabh-msft** [comment]
Looks like there are few merge conflicts.. Can you please pull latest code

File: schema/swa-cli.config.schema.json [hunk 2]
```diff
@@ -264,6 +264,14 @@
         "dryRun": {
           "description": "Simulate a deploy process without actually running it",
           "type": "boolean"
+        },
+        "apiLanguage": {
+          "description": "the runtime language of the function",
+          "type": "string"
+        },
+        "apiVersion": {
+          "description": "version of the function runtime language",
```
Comment (sulabh-msft) [inline schema/swa-cli.config.schema.json:273]: nit: Shall we start with Caps. Please check at other places in this file

File: src/cli/commands/deploy.ts (hunk 1 above)
Comment (sulabh-msft) [reply src/cli/commands/deploy.ts:34]: +1. @sgollapudi77  Lets include those changes here.

**Review by sulabh-msft — CHANGES_REQUESTED** [review]

File: src/cli/commands/deploy.ts (hunk 1 above)
Comment (sgollapudi77) [reply src/cli/commands/deploy.ts]: Added these now.

**Review by manekinekko — APPROVED** [review]

**Comment by manekinekko** [comment]
@sgollapudi77 thank you for adding these changes. Could you also update the [docs ](https://azure.github.io/static-web-apps-cli/docs/use/deploy#53-deploy-a-front-end-app-with-an-api)to reflect these changes?

**Comment by sgollapudi77** [comment]
Closing this PR and re-opening new one against main branch.

**Pull request closed without merging**